ops/framework.py

```diff
@@ -906,11 +906,11 @@ def _reemit(self, single_event_path: Optional[str] = None):
                 if single_event_path is None:
                     logger.debug("Re-emitting deferred event %s.", event)
                 elif isinstance(event, LifecycleEvent):
-                    # Ignore Lifecycle events: they are "private" and not interesting.
-                    pass
+                    logger.debug("Emitting lifecycle event %s.", event.handle.kind)
```

Comment:
I think the change of parent class to `LifecycleEvent` is reasonable, however I agree with your hesitation -- I think this is going to be too much logging.

I don't really think it needs to be logged at all, even for collect-status events, and the log message doesn't really contain much useful information that you can't see by looking at the charm statically anyway. So I'd say just leave this as ignore/pass.

Reply:
My reason for wanting to log collect-status is that now the collect-status event handler has (potential for) holding charm logic as it is user-facing. And the fact that leader/followers will get different events would be nice to show in jhack tail.

I haven't used the event extensively (yet) in practice to know whether it'd be useful, it's just a hunch.

Would be nice to gather feedback from other charmers who are trying it out?

Reply:
So is the concern primarily not the debug log itself, but that this will mean it will show up in jhack tail?

As far as the log itself goes, for charms that use collect-status it'll mean we get two (or three) logs for every Juju event, which I don't think is good. I think even when debugging with jhack, you'll know the charm uses collect-status, so you can just assume that `collect_unit_status` is occurring when you see any other event. And the log doesn't add any other useful collect_status-specific information.

In any case, I think we should start by not logging it for now (that is, reinstate the "Ignore Lifecycle events" comment), and if we *collect* more information about why it'd be useful, we can reconsider.